src/components/ReleaseMonitor/ApplicationFilter.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import {
+  Menu,
+  MenuContent,
+  MenuList,
+  MenuItem,
+  MenuToggle,
+  Badge,
+  MenuContainer,
+} from '@patternfly/react-core';
+import { FilterIcon } from '@patternfly/react-icons/dist/esm/icons/filter-icon';
+
+interface ApplicationFilterProps {
+  applications: string[];
+  onSelectionsChange: (selections: string[]) => void;
+}
+
+export const ApplicationFilter: React.FC<ApplicationFilterProps> = ({
+  applications,
+  onSelectionsChange,
+}) => {
+  const [selections, setSelections] = React.useState<string[]>([]);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const toggleRef = React.useRef<HTMLButtonElement>(null);
+  const menuRef = React.useRef<HTMLDivElement>(null);
+  const containerRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    const handleMenuKeys = (event: KeyboardEvent) => {
+      if (isOpen && menuRef.current?.contains(event.target as Node)) {
+        if (event.key === 'Escape' || event.key === 'Tab') {
+          setIsOpen(!isOpen);
+          toggleRef.current?.focus();
+        }
+      }
+    };
+
+    const handleClickOutside = (event: MouseEvent) => {
+      if (isOpen && !menuRef.current?.contains(event.target as Node)) {
+        setIsOpen(false);
+      }
+    };
+    window.addEventListener('keydown', handleMenuKeys);
+    window.addEventListener('click', handleClickOutside);
+    return () => {
+      window.removeEventListener('keydown', handleMenuKeys);
+      window.removeEventListener('click', handleClickOutside);
+    };
+  }, [isOpen, setIsOpen]);
```

Comment:
MenuContainer component supports these by default. No need to implement these explicitly.